homework18/js/main.js

```diff
@@ -0,0 +1,196 @@
+                    let filmId = event.target.id;
+                    getDetails(filmId);
+                }
+            })
+        }
+    } catch (error) {
+        console.error(error);
+    }
+}
+
+//Будуємо блок та показуємо детальну інформацію про фільм
+const showDetails = (details) => {
+    document.querySelector('.film-details').innerHTML = '';
+    const filmPoster = document.createElement('img');
+    if(details.Poster === 'N/A') {
+        filmPoster.src = '../assets/poster.jpg';
+    } else {
+        filmPoster.src = `${details.Poster}`;
+    }
+    filmPoster.classList.add('film-details__poster');
+    filmPoster.alt = `Poster of "${details.Title}"`;
+
+    const title = document.createElement('p');
+    title.innerHTML = `<span class="characteristic film-details__title">${details.Title} </span>`;
+    const year = document.createElement('p');
+    year.innerHTML = `<span class="characteristic"> Year: </span> ${details.Year}`;
+
+    const country = document.createElement('p');
+    country.innerHTML = `<span class="characteristic"> Country: </span> ${details.Country}`;
+
+    const director = document.createElement('p');
+    director.innerHTML = `<span class="characteristic"> Director: </span> ${details.Director}`;
+
+    const writer = document.createElement('p');
+    writer.innerHTML = `<span class="characteristic"> Writer: </span> ${details.Writer}`;
+
+    const type = document.createElement('p');
+    type.innerHTML = `<span class="characteristic"> Type: </span> ${details.Type}`;
+
+    const genre = document.createElement('p');
+    genre.innerHTML = `<span class="characteristic"> Genre: </span> ${details.Genre}`;
+
+    const language = document.createElement('p');
+    language.innerHTML = `<span class="characteristic"> Language: </span> ${details.Language}`;
+
+    const released = document.createElement('p');
+    released.innerHTML = `<span class="characteristic"> Released: </span> ${details.Released}`;
+
+    const runtime = document.createElement('p');
+    runtime.innerHTML = `<span class="characteristic"> Runtime: </span> ${details.Runtime}`;
+
+    const actors = document.createElement('p');
+    actors.innerHTML = `<span class="characteristic"> Actors: </span> ${details.Actors}`;
+
+    const awards = document.createElement('p');
+    awards.innerHTML = `<span class="characteristic"> Awards: </span> ${details.Awards}`;
+
+    const plot = document.createElement('p');
+    plot.innerHTML = `<span class="characteristic__plot"> <span class="characteristic"> Plot: </span><br> ${details.Plot} </span>`;
+
+    const description = document.createElement('div');
+    description.classList.add('film-details__content');
+    description.append(title, year, country, director, writer, type, genre, language, released, runtime, actors, awards, plot);
+
+    const divDetails =  document.querySelector('.film-details');
+    divDetails.append(filmPoster, description);
```

Comment:
This can be optimised using less lined of code. For example create an array `= ['title', 'year', 'country',...]` and map through it

Each time you will `document.createElement('p');` and `element.innerHTML = `<span class="characteristic"> Awards: </span> ${details.Awards}``